pkg/linkdiscovery/connection.go

```diff
@@ -171,6 +172,16 @@ func (c *Controller) processPacket(packetIn *p4api.PacketIn) {
 		}
 		c.updateIngressLink(pim.IngressPort, uint32(egressPort), string(lldp.ChassisID.ID))
 	}
+
+	// if condition to process ARP packet
+	arpLayer := rawPacket.Layer(layers.LayerTypeARP)
+	if arpLayer != nil {
+		pim := c.codec.DecodePacketInMetadata(packetIn.Metadata)
+		arp := arpLayer.(*layers.ARP)
+		// FIXME: pim.RoleAgentID substitutes device.ID - is it correct? Should I abandon it?
```

Comment:
I would not worry about this here.  We don't need this.

Reply:
Comment was removed